discovery-infra/test_infra/logs_utils.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import os
+import shutil
+import tarfile
+import time
+from tempfile import TemporaryDirectory
+
+from test_infra.consts import NUMBER_OF_MASTERS
+
+
+def verify_logs_uploaded_and_delete(cluster_tar_path, expected_min_log_num, installation_success):
```

Comment:
The only thing that this function does and `verify_logs_uploaded` doesn't is removing `cluster_tar_path` - it is redundant IMO.

Lets just keep it simple and use only `verify_logs_uploaded` and delete `cluster_tar_path` in it as it was originally.

It is not a big deal to download the tar on each try and it is state less - so simpler and more robust.